packages/orm/src/relations/OneToOneReference.ts

```diff
@@ -124,8 +124,8 @@ export class OneToOneReferenceImpl<T extends Entity, U extends Entity>
   async load(opts: { withDeleted?: boolean; forceReload?: boolean } = {}): Promise<U | undefined> {
     ensureNotDeleted(this.entity, { ignore: "pending" });
     if (!this._isLoaded || opts.forceReload) {
-      if (!this.entity.isNewEntity) {
-        this.loaded = await oneToOneDataLoader(this.entity.em, this).load(this.entity.idOrFail);
+      if (!this.entity.isNewEntity && this.entity.idTagged) {
+        this.loaded = await oneToOneDataLoader(this.entity.em, this).load(this.entity.idTagged);
```

Comment:
@stephenh can i get thoughts on this

Reply:
Hm, @mmahalwy yeah, looks like we should add an `idTaggedOrFail`.

I was futzing with adding a type-guard is `isNewEntity`, i.e. within an `if (isNewEntity)` check, ideally we could narrow the type of `idTagged` to just string, but it looks like TypeScript doesn't support type guards on getters, which is odd... :shrug: 

I suppose we could also do an `idTagged!`, with the hope that eventually TS will let us do the type guard approach, but dunno, I think probably `idTaggedOrFail` is best for now...

Reply:
im cool with `idTaggedOrFail`. Should be an easy add. LMK

Reply:
Cool, yep, let's do that. Thanks!

Reply:
ill add it to this PR